src/api.ts

```diff
@@ -48,43 +25,26 @@ export const FACTORIES: DTODecoratorFactories = {
     IsEnum,
     IsInteger,
     IsNested,
+    IsNumber,
     IsString,
     IsUUID,
 };
 
-/* The factories can be transformed into an alternate implementation.
- */
-export type DTODecoratorTransformers = {
-    [Key in keyof DTODecoratorFactories]?: PropertyDecoratorTransformer<unknown>;
-};
-
-/* Applying a transformer to a factory produces a new factory.
- */
-export function applyTransformer<Options>(
-    factory: PropertyDecoratorFactory<Options>,
-    transformer?: PropertyDecoratorTransformer<Options>,
-): PropertyDecoratorFactory<Options> {
-    return (options: Options) => {
-        const decorator = factory(options);
-        return transformer !== undefined ? transformer(options, decorator) : decorator;
-    };
-}
-
-/* An implementation can produce its own factories using `buildDecoratorFactories` to map over the factories.
+/* Consumers that wish to define their own behavior may compose sets of these decorator factories together.
  */
-export function buildDecoratorFactories(
-    transformers: DTODecoratorTransformers,
-    decorators: DTODecoratorFactories = FACTORIES,
+export function composeDecoratorFactories(
```

Comment:
The main API has changed to a composition function.

I intend to follow up with two changes:
 1. Make the reflect-metadata decorators non-default. The default factories should probably return a noop decorator.
 2. Refactor the deprecation warning factories to look a bit more like the other code.